### ISSUE TITLE
Change model.fit() to accept Numpy arrays X, y,... as default.

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#1036]
Currently, model.fit() requires a dc.data object as input. This contrasts with sklearn and keras which have defaults model.fit(X, y). I propose that we change our fit() method to follow the same API.

The main advantage of this change is that we could onboard new users more gently and remove a common gotcha. We also get better interoperability with the broader python ML ecosystem.

For backwards compatibility, we can add a check in the next version that looks for the old input style and issues a warning. Eventually, in case users want to pass in a dc.data object, they will need to set a keyword argument.

**Comment by peastman** [comment]
TensorGraph already has a `fit_on_batch()` method that takes numpy arrays.  Contrary to its name, it does *not* require the arrays to be a single batch.  It just constructs a NumpyDataset from them and passes it on to the regular `fit()` method.  The only hardcoded assumption in that method is that there's only a single epoch, and that could easily be changed.

This seems like part of the discussion about long term architecture, harmonizing dc.data with tf.data and TensorGraph layers with Keras layers.  Do we ultimately want to have a single `fit()` method that can accept any sort of object, whether numpy arrays or a DeepChem dataset or a Tensorflow dataset?

**Comment by rbharath** [comment]
Agreed this is part of the longer term architecture discussion. My instinct is that we should optimize for interoperability with the ecosystem. This will enable us to gain more users and be more useful in general.

I think supporting a single comprehensive fit() might work well since it will "just work" for most people.

**Comment by rbharath** [comment]
Closing in favor of #1795. Let's continue this discussion there.
